### PR TITLE
Add Item class 

### PR DESCRIPTION
**Pull request by joannasese**
Add Item class to Java Client Library

To create an item:

```java
final Item item = new Item();
item.setItemCode("iron_maidens");
item.setName("Iron Maidens");
client.createItem(item);
```

To look up an item (item_code required):

```java
final Item item = client.getItem("iron_maiden");
```

To update an item (item_code required):

```java
final Item item = new Item();
item.setItemCode("powder_puff");
item.setName("Powder Puff");
client.createItem(item);

final CustomFields customFields = new CustomFields();
final CustomField field = new CustomField();
field.setName("size");
field.setValue("small");
customFields.add(field);

item.setCustomFields(customFields);

final Item updatedItem = client.updateItem(item);
```

To delete an item (item_code required):

```java
client.deleteItem("powder_puff");
```

To restore an item (item required):

```java
final Item item = client.getItem("iron_maiden");
client.reactivateItem(item);
```

To create an adjustment including an item_code:

```java
final Adjustment a = new Adjustment();
a.setUnitAmountInCents(150);
a.setCurrency("USD");
a.setItemCode("gray_it");
a.setEndDate("2020-08-03T16:02:24Z");

client.createAccountAdjustment(account.getAccountCode, a);
```

### CODE REVIEW

File: src/main/java/com/ning/billing/recurly/RecurlyClient.java [hunk 1]
```diff
@@ -1372,6 +1374,79 @@ public Transaction enterOfflinePayment(final String invoiceId, final Transaction
...
+    }
+
+    /**
+     * Update an Item's info
+     * <p>
+     *
+     * @param item The Item to update on recurly
+     * @return the updated item object
+     */
+    public Item updateItem(final Item item) {
```
Comment (bhelx) [inline src/main/java/com/ning/billing/recurly/RecurlyClient.java]: I think you should separate out the id and the object here like we do in the other update methods. Signature would look like this: `updateItem(final String itemId, final Item item)`. The reason is you aren't necessarily updating the item being passed in, it's just acting as a request body.

File: src/main/java/com/ning/billing/recurly/RecurlyClient.java [hunk 2]
```diff
@@ -1372,6 +1374,79 @@ public Transaction enterOfflinePayment(final String invoiceId, final Transaction
...
+
+    /**
+     * Reactivating a canceled item
+     * <p>
+     * Reactivate a canceled item.
+     *
+     * @param item Item object
+     * @return Item
+     */
+    public Item reactivateItem(final Item item) {
```
Comment (bhelx) [inline src/main/java/com/ning/billing/recurly/RecurlyClient.java]: This should take an an itemCode String, not an Item

**Comment by joannasese** [comment]
@pierre good day! when you have a second, would you mind publishing a release? thank you. 🙏

**Comment by pierre** [comment]
0.32.0 has been released. Happy holidays! 🦃

**Pull request merged**